.github/workflows/ci.yml

```diff
@@ -0,0 +1,58 @@
+name: Tests
+
+on: [push]
+
+jobs:
+  test:
+    name: ${{ matrix.os }} (R ${{ matrix.r-version }})
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+            r-version: '3.6'
+          - os: ubuntu-latest
+            r-version: 'release'
+          - os: macos-latest
+            r-version: '3.6'
+          - os: macos-latest
+            r-version: 'release'
+    steps:
+      - name: checkout repository
+        uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 1
+      - name: set up R
+        uses: r-lib/actions/setup-r@v1
+        with:
+          r-version: ${{ matrix.r-version }}
+      - uses: r-lib/actions/setup-pandoc@v1
+      - uses: r-lib/actions/setup-tinytex@v1
+      - name: set up dependencies
+        shell: bash
+        run: |
+          if [[ "${{ matrix.os }}" == "macos-latest" ]]; then
+              export OS_NAME="macos"
+          elif [[ "${{ matrix.os }}" == "ubuntu-latest" ]]; then
+              export OS_NAME="linux"
+          fi
+          $GITHUB_WORKSPACE/.github/setup.sh
+      - name: run tests
+        shell: bash
+        run: |
+          $GITHUB_WORKSPACE/test.sh
+      - name: upload coverage
+        if: ${{ matrix.os == 'macos-latest' && matrix.r-version == 'release' }}
+        shell: bash
```

Comment:
As just one general suggestion, if we don't need to run bash code, you can potentially simplify stuff by setting

```
shell: Rscript {0}
```

and just writing R code under `run` directly.

Reply:
nice touch! ok I'll do that